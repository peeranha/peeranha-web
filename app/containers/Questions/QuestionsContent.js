import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

import { getTimeFromDateToNow } from 'utils/datetime';
import * as routes from 'routes-config';

import messages from './messages';

const QuestionItem = item => (
  <div className="question-item">
    <div className="title-user">
      <Link
        className="highlighted-link"
        to={routes.question_view(item.id)}
        href={routes.question_view(item.id)}
      >
        <h5 className="highlighted-link">{item.title}</h5>
      </Link>
      <p>
        <Link
          className="highlighted-link"
          to={routes.profile_view(item.user)}
          href={routes.profile_view(item.user)}
        >
          {item.user}
        </Link>
        {' | '}
        <span>{`${getTimeFromDateToNow(item.post_time, item.locale)} `}</span>
        <FormattedMessage {...messages.ago} />
      </p>
    </div>
    <div className="votes-answers">
      <div className="votes">
        <p className="number">{item.rating}</p>
        <p>
          <FormattedMessage {...messages.votes} />
        </p>
      </div>
      <div className="answers" data-bg={!!item.correct_answer_id}>
        <p className="number">{item.answers.length}</p>
        <p>
          <FormattedMessage {...messages.answers} />
        </p>
      </div>
    </div>
  </div>
);

const QuestionsContent = props => (
  <div>
    {props.questionsList.map(item => (
      <QuestionItem {...item} locale={props.locale} key={item.id} />
    ))}
  </div>
);

QuestionsContent.propTypes = {
  questionsList: PropTypes.object.isRequired,
  locale: PropTypes.string.isRequired,
};

export { QuestionItem };
export default QuestionsContent;
