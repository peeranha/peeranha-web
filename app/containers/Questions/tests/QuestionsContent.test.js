import QuestionsContent, { QuestionItem } from '../QuestionsContent';

const props = {
  questionsList: [
    {
      answers: [],
      id: 1,
    },
  ],
  locale: 'en',
};

describe('QuestionsContent', () => {
  it('QuestionsContent, snapshot test', () => {
    expect(QuestionsContent(props)).toMatchSnapshot();
  });

  it('QuestionItem, snapshot test', () => {
    const obj = {
      answers: [],
    };
    expect(QuestionItem(obj)).toMatchSnapshot();
  });
});
