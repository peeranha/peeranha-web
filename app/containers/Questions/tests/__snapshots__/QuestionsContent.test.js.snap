// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsContent QuestionItem, snapshot test 1`] = `
<Link
  href="/questions/undefined"
  replace={false}
  to="/questions/undefined"
>
  <div
    className="question-item"
  >
    <div
      className="title-user"
    >
      <Link
        className="highlighted-link"
        href="/questions/undefined"
        replace={false}
        to="/questions/undefined"
      >
        <h5
          className="highlighted-link"
        />
      </Link>
      <p>
        <Link
          className="highlighted-link"
          href="/users/undefined"
          replace={false}
          to="/users/undefined"
        />
         | 
        <span>
          almost NaN years
        </span>
      </p>
    </div>
    <div
      className="votes-answers"
    >
      <div
        className="votes"
      >
        <p
          className="number"
        />
        <p>
          <FormattedMessage
            defaultMessage="votes"
            id="app.containers.Questions.votes"
            values={Object {}}
          />
        </p>
      </div>
      <div
        className="answers"
        data-bg={false}
      >
        <p
          className="number"
        >
          0
        </p>
        <p>
          <FormattedMessage
            defaultMessage="answers"
            id="app.containers.Questions.answers"
            values={Object {}}
          />
        </p>
      </div>
    </div>
  </div>
</Link>
`;

exports[`QuestionsContent QuestionsContent, snapshot test 1`] = `
<div>
  <QuestionItem
    answers={Array []}
    id={1}
    locale="en"
  />
</div>
`;
